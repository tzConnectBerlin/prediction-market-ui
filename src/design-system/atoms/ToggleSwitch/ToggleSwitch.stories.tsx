import * as React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import { ToggleSwitch, ToggleProps } from './ToggleSwitch';

export default {
  title: 'Atom/ToggleSwitch',
  component: ToggleSwitch,
} as Meta;

const Template: Story<ToggleProps> = (args) => <ToggleSwitch {...args} />;

export const ToggleSwitchWithoutTooltip = Template.bind({});
ToggleSwitchWithoutTooltip.args = {
  label: 'Advanced View',
};

export const ToggleSwitchWithTooltip = Template.bind({});
ToggleSwitchWithTooltip.args = {
  label: 'Advanced View',
  tooltip: true,
};
