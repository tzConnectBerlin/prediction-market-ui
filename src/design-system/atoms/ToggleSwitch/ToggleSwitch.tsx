import React from 'react';
import Switch from '@material-ui/core/Switch';
import FormGroup from '@material-ui/core/FormGroup';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import styled from '@emotion/styled';
import { Theme, useTheme } from '@material-ui/core';
import { CustomInputLabel } from '../../molecules/CustomInputLabel';

const StyledFormGroup = styled(FormGroup)`
  &.MuiFormGroup-root {
    padding-right: 1rem;
  }
`;

const StyledLabel = styled(FormControlLabel)<{ theme: Theme }>`
  &.MuiFormControlLabel-root {
    color: ${({ theme }) => theme.palette.primary.main};
    justify-content: space-between;
  }

  & .MuiFormControlLabel-label {
    font-size: 0.75rem;
    font-weight: 700;
    font-family: Roboto Mono;
  }

  & .MuiSwitch-root {
  }
`;

export interface ToggleProps {
  label: string;
}

export const ToggleSwitch: React.FC<ToggleProps> = ({ label }) => {
  const [checked, setChecked] = React.useState(false);

  const toggleChecked = () => {
    setChecked((prev) => !prev);
  };
  const theme = useTheme();

  return (
    <StyledFormGroup>
      <StyledLabel
        control={<Switch size="medium" checked={checked} onChange={toggleChecked} />}
        label={label}
        labelPlacement="start"
        theme={theme}
      />
    </StyledFormGroup>
  );
};
