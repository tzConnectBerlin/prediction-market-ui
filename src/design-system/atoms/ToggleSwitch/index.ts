export { ToggleSwitch, ToggleSwitch as Default } from './ToggleSwitch';
export type { ToggleProps } from './ToggleSwitch';
