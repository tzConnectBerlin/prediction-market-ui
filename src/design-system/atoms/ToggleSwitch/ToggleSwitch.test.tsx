import renderer from 'react-test-renderer';
import { ToggleSwitch } from './ToggleSwitch';

describe('Snapshot testing ToggleSwitch Component', () => {
  it('renders correctly with tooltip', () => {
    const tooltip = renderer.create(<ToggleSwitch label="Advanced View" tooltip />).toJSON();
    expect(tooltip).toMatchSnapshot();
  });

  it('renders correctly without tooltip', () => {
    const tooltip = renderer.create(<ToggleSwitch label="Advanced View" />).toJSON();
    expect(tooltip).toMatchSnapshot();
  });
});
