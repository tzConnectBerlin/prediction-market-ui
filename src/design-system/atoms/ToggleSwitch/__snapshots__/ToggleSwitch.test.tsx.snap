// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing ToggleSwitch Component renders correctly with tooltip 1`] = `
.emotion-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  color: #1976d2;
  margin-left: 1rem;
  padding-right: 1.5rem;
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.75;
  letter-spacing: 0.00938em;
  color: #1976d2;
}

.emotion-2.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-3 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  margin-left: 0.5rem;
}

.emotion-3.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0, 0, 0, 0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-4::-moz-focus-inner {
  border-style: none;
}

.emotion-4.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-4 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-4:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-4:hover {
    background-color: transparent;
  }
}

.emotion-4.Mui-disabled {
  background-color: transparent;
  color: rgba(0, 0, 0, 0.26);
}

.emotion-5 {
  border-radius: 50%;
  background: rgba(1, 102, 255, 0.16);
  height: 0.7em;
  width: 0.7em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: 58px;
  height: 38px;
  overflow: hidden;
  padding: 12px;
  box-sizing: border-box;
  position: relative;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  z-index: 0;
  vertical-align: middle;
}

@media print {
  .emotion-7 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  padding: 9px;
  border-radius: 50%;
  position: absolute;
  top: 0;
  left: 0;
  z-index: 1;
  color: #fff;
  -webkit-transition: left 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,-webkit-transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: left 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-8::-moz-focus-inner {
  border-style: none;
}

.emotion-8.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-8 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-8.Mui-checked {
  -webkit-transform: translateX(20px);
  -moz-transform: translateX(20px);
  -ms-transform: translateX(20px);
  transform: translateX(20px);
}

.emotion-8.Mui-disabled {
  color: #f5f5f5;
}

.emotion-8.Mui-checked+.MuiSwitch-track {
  opacity: 0.5;
}

.emotion-8.Mui-disabled+.MuiSwitch-track {
  opacity: 0.12;
}

.emotion-8 .MuiSwitch-input {
  left: -100%;
  width: 300%;
}

.emotion-8:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-8:hover {
    background-color: transparent;
  }
}

.emotion-8.Mui-checked {
  color: #1976d2;
}

.emotion-8.Mui-checked:hover {
  background-color: rgba(25, 118, 210, 0.04);
}

@media (hover: none) {
  .emotion-8.Mui-checked:hover {
    background-color: transparent;
  }
}

.emotion-8.Mui-checked.Mui-disabled {
  color: rgb(167, 202, 237);
}

.emotion-8.Mui-checked+.MuiSwitch-track {
  background-color: #1976d2;
}

.emotion-9 {
  cursor: inherit;
  position: absolute;
  opacity: 0;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  margin: 0;
  padding: 0;
  z-index: 1;
}

.emotion-10 {
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  background-color: currentColor;
  width: 20px;
  height: 20px;
  border-radius: 50%;
}

.emotion-11 {
  height: 100%;
  width: 100%;
  border-radius: 7px;
  z-index: -1;
  -webkit-transition: opacity 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  background-color: #000;
  opacity: 0.38;
}

<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined MuiGrid-wrap-xs-nowrap emotion-0"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-undefined emotion-1"
  >
    <h6
      className="MuiTypography-root MuiTypography-subtitle1 emotion-2"
      lines={3}
    >
      Advanced View
    </h6>
    <p
      className="MuiTypography-root MuiTypography-body1  emotion-3"
      lines={3}
    >
      <button
        aria-label="tooltip"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium emotion-4"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <div
          className="emotion-5"
        >
          <svg
            fill="#1976d2"
            height="0.7em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="0.7em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M232 235h48v137h-48zM232 140h48v48h-48z"
            />
          </svg>
        </div>
      </button>
    </p>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item emotion-6"
  >
    <span
      className="MuiSwitch-root MuiSwitch-sizeMedium emotion-7"
    >
      <span
        className="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root emotion-8"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <input
          checked={false}
          className="MuiSwitch-input PrivateSwitchBase-input emotion-9"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="MuiSwitch-thumb emotion-10"
        />
      </span>
      <span
        className="MuiSwitch-track emotion-11"
      />
    </span>
  </div>
</div>
`;

exports[`Snapshot testing ToggleSwitch Component renders correctly without tooltip 1`] = `
.emotion-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  color: #1976d2;
  margin-left: 1rem;
  padding-right: 1.5rem;
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.75;
  letter-spacing: 0.00938em;
  color: #1976d2;
}

.emotion-2.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: 58px;
  height: 38px;
  overflow: hidden;
  padding: 12px;
  box-sizing: border-box;
  position: relative;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  z-index: 0;
  vertical-align: middle;
}

@media print {
  .emotion-4 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  padding: 9px;
  border-radius: 50%;
  position: absolute;
  top: 0;
  left: 0;
  z-index: 1;
  color: #fff;
  -webkit-transition: left 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,-webkit-transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: left 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-5::-moz-focus-inner {
  border-style: none;
}

.emotion-5.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-5 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-5.Mui-checked {
  -webkit-transform: translateX(20px);
  -moz-transform: translateX(20px);
  -ms-transform: translateX(20px);
  transform: translateX(20px);
}

.emotion-5.Mui-disabled {
  color: #f5f5f5;
}

.emotion-5.Mui-checked+.MuiSwitch-track {
  opacity: 0.5;
}

.emotion-5.Mui-disabled+.MuiSwitch-track {
  opacity: 0.12;
}

.emotion-5 .MuiSwitch-input {
  left: -100%;
  width: 300%;
}

.emotion-5:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-5:hover {
    background-color: transparent;
  }
}

.emotion-5.Mui-checked {
  color: #1976d2;
}

.emotion-5.Mui-checked:hover {
  background-color: rgba(25, 118, 210, 0.04);
}

@media (hover: none) {
  .emotion-5.Mui-checked:hover {
    background-color: transparent;
  }
}

.emotion-5.Mui-checked.Mui-disabled {
  color: rgb(167, 202, 237);
}

.emotion-5.Mui-checked+.MuiSwitch-track {
  background-color: #1976d2;
}

.emotion-6 {
  cursor: inherit;
  position: absolute;
  opacity: 0;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  margin: 0;
  padding: 0;
  z-index: 1;
}

.emotion-7 {
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  background-color: currentColor;
  width: 20px;
  height: 20px;
  border-radius: 50%;
}

.emotion-8 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-9 {
  height: 100%;
  width: 100%;
  border-radius: 7px;
  z-index: -1;
  -webkit-transition: opacity 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  background-color: #000;
  opacity: 0.38;
}

<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined MuiGrid-wrap-xs-nowrap emotion-0"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-undefined emotion-1"
  >
    <h6
      className="MuiTypography-root MuiTypography-subtitle1 emotion-2"
      lines={3}
    >
      Advanced View
    </h6>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item emotion-3"
  >
    <span
      className="MuiSwitch-root MuiSwitch-sizeMedium emotion-4"
    >
      <span
        className="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root emotion-5"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <input
          checked={false}
          className="MuiSwitch-input PrivateSwitchBase-input emotion-6"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="MuiSwitch-thumb emotion-7"
        />
        <span
          className="MuiTouchRipple-root emotion-8"
        />
      </span>
      <span
        className="MuiSwitch-track emotion-9"
      />
    </span>
  </div>
</div>
`;
